test/simple_sliding_window_test.rb

```diff
@@ -0,0 +1,67 @@
+require 'test_helper'
+
+class TestSimpleSlidingWindow < MiniTest::Unit::TestCase
+  CLASS = ::Semian::Simple::SlidingWindow
+
+  def setup
+    @sliding_window = CLASS.new(max_size: 6)
+    @sliding_window.clear
+  end
+
+  def teardown
+    @sliding_window.destroy
+  end
+
+  module SlidingWindowTestCases
+    def test_sliding_window_push
+      assert_equal(0, @sliding_window.size)
+      @sliding_window << 1
+      assert_correct_first_and_last_and_size(@sliding_window, 1, 1, 1, 6)
+      @sliding_window << 5
+      assert_correct_first_and_last_and_size(@sliding_window, 1, 5, 2, 6)
+    end
+
+    def test_sliding_window_resize
+      assert_equal(0, @sliding_window.size)
+      @sliding_window << 1 << 2 << 3 << 4 << 5 << 6
+      assert_correct_first_and_last_and_size(@sliding_window, 1, 6, 6, 6)
+      @sliding_window.resize_to 6
+      assert_correct_first_and_last_and_size(@sliding_window, 1, 6, 6, 6)
+      @sliding_window.resize_to 5
+      assert_correct_first_and_last_and_size(@sliding_window, 2, 6, 5, 5)
+      @sliding_window.resize_to 6
+      assert_correct_first_and_last_and_size(@sliding_window, 2, 6, 5, 6)
+    end
```

Comment:
This is the only case where you use `resize_to`—it's not actually used in the circuit breaker, so why not remove it altogether?